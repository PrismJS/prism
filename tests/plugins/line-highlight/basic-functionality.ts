import { createTestSuite } from '../../helper/prism-dom-util';

// just a few lines of JS code, so we have something to highlight
const exampleCode = String.raw`// foo
var foo = 0;
var bar = 1;
var baz = foo + bar;
function add(a, b) {
	return a + b;
}
const sub = (a, b) => a - b;
`;

describe('Line Highlight', () => {
	const { it } = createTestSuite({
		languages: 'javascript',
		plugins: 'line-highlight'
	});

	it('single line', ({ util }) => {
		util.assert.highlightPreElement({
			language: 'none',
			code: exampleCode,
			attributes: { 'data-line': '2' },
		});
	});

	it('single range', ({ util }) => {
		util.assert.highlightPreElement({
			language: 'none',
			code: exampleCode,
			attributes: { 'data-line': '3-5' },
		});
	});

	it('multiple ranges', ({ util }) => {
		util.assert.highlightPreElement({
			language: 'none',
			code: exampleCode,
			attributes: { 'data-line': '1, 3-5, 8' },
		});
	});

	// Fixed by https://github.com/PrismJS/prism/pull/3518
	it('offset', ({ util }) => {
		util.assert.highlightPreElement({
			language: 'none',
			code: exampleCode,
			attributes: { 'data-line': '42', 'data-line-offset': '40' },
		});
		util.assert.highlightPreElement({
			language: 'none',
			code: exampleCode,
			attributes: { 'data-line': '41, 43-45, 48', 'data-line-offset': '40' },
		});
	});

	it('out of bounds ranges', ({ util }) => {
		util.assert.highlightPreElement({
			language: 'none',
			code: exampleCode,
			attributes: { 'data-line': '400' },
		});
		util.assert.highlightPreElement({
			language: 'none',
			code: exampleCode,
			attributes: { 'data-line': '4-400' },
		});
	});
});
