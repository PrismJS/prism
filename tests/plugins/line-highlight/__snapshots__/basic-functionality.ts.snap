// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Line Highlight multiple ranges 1`] = `
"<pre data-line=\\"1, 3-5, 8\\" class=\\"language-none\\" tabindex=\\"0\\"><code class=\\"language-none\\">// foo
var foo = 0;
var bar = 1;
var baz = foo + bar;
function add(a, b) {
	return a + b;
}
const sub = (a, b) =&gt; a - b;
<div aria-hidden=\\"true\\" data-range=\\"1\\" class=\\" line-highlight\\" data-start=\\"1\\" style=\\"width: 0px;\\"> 
</div><div aria-hidden=\\"true\\" data-range=\\"3-5\\" class=\\" line-highlight\\" data-start=\\"3\\" data-end=\\"5\\" style=\\"width: 0px;\\"> 
 
 
</div><div aria-hidden=\\"true\\" data-range=\\"8\\" class=\\" line-highlight\\" data-start=\\"8\\" style=\\"width: 0px;\\"> 
</div></code></pre>"
`;

exports[`Line Highlight offset 1`] = `
"<pre data-line=\\"42\\" data-line-offset=\\"40\\" class=\\"language-none\\" tabindex=\\"0\\"><code class=\\"language-none\\">// foo
var foo = 0;
var bar = 1;
var baz = foo + bar;
function add(a, b) {
	return a + b;
}
const sub = (a, b) =&gt; a - b;
<div aria-hidden=\\"true\\" data-range=\\"42\\" class=\\" line-highlight\\" data-start=\\"42\\" style=\\"width: 0px;\\"> 
</div></code></pre>"
`;

exports[`Line Highlight offset 2`] = `
"<pre data-line=\\"41, 43-45, 48\\" data-line-offset=\\"40\\" class=\\"language-none\\" tabindex=\\"0\\"><code class=\\"language-none\\">// foo
var foo = 0;
var bar = 1;
var baz = foo + bar;
function add(a, b) {
	return a + b;
}
const sub = (a, b) =&gt; a - b;
<div aria-hidden=\\"true\\" data-range=\\"41\\" class=\\" line-highlight\\" data-start=\\"41\\" style=\\"width: 0px;\\"> 
</div><div aria-hidden=\\"true\\" data-range=\\"43-45\\" class=\\" line-highlight\\" data-start=\\"43\\" data-end=\\"45\\" style=\\"width: 0px;\\"> 
 
 
</div><div aria-hidden=\\"true\\" data-range=\\"48\\" class=\\" line-highlight\\" data-start=\\"48\\" style=\\"width: 0px;\\"> 
</div></code></pre>"
`;

exports[`Line Highlight out of bounds ranges 1`] = `
"<pre data-line=\\"400\\" class=\\"language-none\\" tabindex=\\"0\\"><code class=\\"language-none\\">// foo
var foo = 0;
var bar = 1;
var baz = foo + bar;
function add(a, b) {
	return a + b;
}
const sub = (a, b) =&gt; a - b;
</code></pre>"
`;

exports[`Line Highlight out of bounds ranges 2`] = `
"<pre data-line=\\"4-400\\" class=\\"language-none\\" tabindex=\\"0\\"><code class=\\"language-none\\">// foo
var foo = 0;
var bar = 1;
var baz = foo + bar;
function add(a, b) {
	return a + b;
}
const sub = (a, b) =&gt; a - b;
<div aria-hidden=\\"true\\" data-range=\\"4-400\\" class=\\" line-highlight\\" data-start=\\"4\\" data-end=\\"8\\" style=\\"width: 0px;\\"> 
 
 
 
 
</div></code></pre>"
`;

exports[`Line Highlight single line 1`] = `
"<pre data-line=\\"2\\" class=\\"language-none\\" tabindex=\\"0\\"><code class=\\"language-none\\">// foo
var foo = 0;
var bar = 1;
var baz = foo + bar;
function add(a, b) {
	return a + b;
}
const sub = (a, b) =&gt; a - b;
<div aria-hidden=\\"true\\" data-range=\\"2\\" class=\\" line-highlight\\" data-start=\\"2\\" style=\\"width: 0px;\\"> 
</div></code></pre>"
`;

exports[`Line Highlight single range 1`] = `
"<pre data-line=\\"3-5\\" class=\\"language-none\\" tabindex=\\"0\\"><code class=\\"language-none\\">// foo
var foo = 0;
var bar = 1;
var baz = foo + bar;
function add(a, b) {
	return a + b;
}
const sub = (a, b) =&gt; a - b;
<div aria-hidden=\\"true\\" data-range=\\"3-5\\" class=\\" line-highlight\\" data-start=\\"3\\" data-end=\\"5\\" style=\\"width: 0px;\\"> 
 
 
</div></code></pre>"
`;
